src/math/nthprime.rs

```diff
@@ -0,0 +1,58 @@
+// Generate the nth prime number.
+// Algorithm is inspired by the the optimized version of the Sieve of Eratosthenes.
+pub fn nthprime(nth: u64) -> u64 {
+    let mut total_prime: u64 = 0;
+    let mut size_factor: u64 = 2;
+
+    let mut s: u64 = nth * size_factor;
+    let mut primes: Vec<u64> = Vec::new();
+
+    let n: u64 = nth;
+
+    while total_prime < n {
+        primes = get_primes(s).to_vec();
+
+        total_prime = primes[2..].iter().sum();
+        size_factor += 1;
+        s = n * size_factor;
+    }
+
+    count_prime(primes, n).unwrap()
+}
+
+fn get_primes(s: u64) -> Vec<u64> {
+    let mut v: Vec<u64> = vec![1; s as usize];
```

Comment:
Can you change this to `Vec<bool>`?
It will help with memory usage (and maybe vectorization).

Reply:
Certainly, will do this.

Reply:
Utilizing the vec in the function get_nth_prime. The while loop has a use case for the sum of the count.